bumpversion/__init__.py

```diff
@@ -336,7 +342,62 @@ def __iter__(self):
         return iter(self._values)
 
     def __repr__(self):
-        return '<bumpversion.Version:{}>'.format(keyvaluestring(self._values))
+        by_part = ", ".join("{}={}".format(k, v) for k, v in self.items())
+        return '<bumpversion.Version:{}>'.format(by_part)
+
+    def __hash__(self):
+        return hash(tuple((k, v) for k, v in self.items()))
+
+    def _compare(self, other, method, strict=True):
+        """
+        For strict relations, if a part is equal we skip it
+        :param other: the other Version
+        :param method: the compare method
+        :return:
+        """
+        try:
+            for vals in ((v, other[k]) for k, v in self.items()):
+                if vals[0] == vals[1]:
+                    continue
+                if method(vals[0], vals[1]):
```

Comment:
Can be `return method(vals[0], vals[1])` but then I also prefer to rewrite it a bit:
```python
if vals[0] != vals[1]:
    return method(vals[0], vals[1])
```

Reply:
yes, cant really remember why did I do a if to return true/false.